src/embed.ts

```diff
@@ -336,27 +337,40 @@ class Torus {
     return response;
   }
 
-  async sendTransaction(transaction: Transaction): Promise<string> {
+  async sendTransaction(transaction: TransactionOrVersionedTransaction): Promise<string> {
+    const isVersionedTransaction = isVersionedTransactionInstance(transaction);
+    const message = isVersionedTransaction
+      ? Buffer.from((transaction as VersionedTransaction).serialize()).toString("hex")
+      : (transaction as Transaction).serialize({ requireAllSignatures: false }).toString("hex");
     const response = (await this.provider.request({
       method: "send_transaction",
-      params: { message: transaction.serialize({ requireAllSignatures: false }).toString("hex") },
+      params: { message, isVersionedTransaction },
     })) as string;
     return response;
   }
 
   // support sendOptions
-  async signAndSendTransaction(transaction: Transaction, options?: SendOptions): Promise<{ signature: string }> {
+  async signAndSendTransaction(transaction: TransactionOrVersionedTransaction, options?: SendOptions): Promise<{ signature: string }> {
+    const isVersionedTransaction = isVersionedTransactionInstance(transaction);
+    const message = isVersionedTransaction
+      ? Buffer.from((transaction as VersionedTransaction).serialize()).toString("hex")
+      : (transaction as Transaction).serialize({ requireAllSignatures: false }).toString("hex");
     const response = (await this.provider.request({
       method: "send_transaction",
-      params: { message: transaction.serialize({ requireAllSignatures: false }).toString("hex"), options },
+      params: { message, options, isVersionedTransaction },
```

Comment:
isVersionTransaction should be version like Solanalabs defined
("legacy", "0") 
this to prevent future version 1,2,3 and so on